tests/ui/escape_analysis.rs

```diff
@@ -174,3 +174,8 @@ mod issue_3739 {
         };
     }
 }
+
+/// Issue #5542
+///
+/// This shouldn't warn for `boxed_local` as it is a function implemented in C.
+pub extern "C" fn do_now_warn_me(_c_pointer: Box<String>) -> () {}
```

Comment:
Can we add an additional example omitting the ABI?

Reply:
I'm not sure if I got the idea. Isn't [the `warn_arg` function](https://github.com/rust-lang/rust-clippy/blob/15150c07eaeec50138a9c96c5bdecbdf92c3101e/tests/ui/escape_analysis.rs#L40) already doing it? (See [the expected log](https://github.com/rust-lang/rust-clippy/blob/15150c07eaeec50138a9c96c5bdecbdf92c3101e/tests/ui/escape_analysis.stderr#L4))

Reply:
I mean a function with the exact same signature, but without the `"C"` part. That should default to `"C"`.

It seems rustfmt rewrites that adding the ABI, that was unexpected :) Anyway, I think the test is relevant, so let's ignore rustfmt this time:

```rust
#[rustfmt::skip]
pub extern fn do_not_warn_me2(_c_pointer: Box<String>) -> () {}
```

Reply:
Oh right right, this makes a lot of sense. Gimme a second.